### DISCUSSION TITLE
Sandboxed Firefox and TEMP directory

### DISCUSSION THREAD

**Discussion by tsaost** [sandboxie-plus/Sandboxie#2285] [Documents and notes]
Just a heads-up for people who have problem downloading files using Firefox inside Sandboxie.

TL;DR version: make sure your TEMP and TMP directories are not on a drive/partition that is closed to Sandboxie.

Long explanation:

I have my DefaultBox setup so that my data drives D:\  and E:\ are completely closed to all applications, including Firefox.  Everything is working fine except that I cannot download any files.  Firefox will simply say that the file download "failed", with no further explanation.

So I created a TestBox with the default Sandboxie setting, and found that Firefox has no problem downloading there.

After some trial and error, I found that it was the D:\ drive restriction that caused the problem.  After further testing, I found that it was D:\TEMP that caused the problem.  I have my environment variable TMP and TEMP both set to D:\TEMP.  Changing them to S:\TEMP (S: is a partition I set up specifically to hold Sandboxied data.)

This makes sense because Firefox seems to be always trying to download to the TEMP/TMP directory, even if you already set up a different default download directory.
